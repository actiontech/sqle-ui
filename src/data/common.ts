export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}