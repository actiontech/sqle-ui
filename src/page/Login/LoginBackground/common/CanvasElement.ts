import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
