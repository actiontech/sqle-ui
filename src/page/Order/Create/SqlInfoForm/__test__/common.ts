import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
