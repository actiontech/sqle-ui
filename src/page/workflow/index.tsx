import { Card, Form, Input, PageHeader } from 'antd';
import { PageFormLayout } from '../../data/common';

const Workflow = () => {
  return (
    <>
      <PageHeader title="工单创建" ghost={false}>
        您可以在这里选择数据源进行创建SQL审核工单
      </PageHeader>
      <section className="padding-content">
        <Card title="工单基本信息">
          <Form {...PageFormLayout}>
            <Form.Item name="name" label="工单名称">
              <Input />
            </Form.Item>
          </Form>
        </Card>
      </section>
    </>
  );
};

export default Workflow;
