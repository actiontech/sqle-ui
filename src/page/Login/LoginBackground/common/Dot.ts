import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
