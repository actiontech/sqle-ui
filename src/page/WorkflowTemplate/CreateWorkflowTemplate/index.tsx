import { Button, Card, Result, Row, Typography } from 'antd';
import { AxiosResponse } from 'axios';
import { useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { IWorkFlowStepTemplateReqV1 } from '../../../api/common';
import workflow from '../../../api/workflow';
import { ICreateWorkflowTemplateV1Return } from '../../../api/workflow/index.d';
import EmitterKey from '../../../data/EmitterKey';
import EventEmitter from '../../../utils/EventEmitter';
import WorkflowTemplateForm from '../WorkflowTemplateForm';
import { BaseFormFields } from '../WorkflowTemplateForm/BaseForm/index.type';

const CreateWorkflowTemplate = () => {
  const { t } = useTranslation();
  const baseFormValue = useRef<BaseFormFields>();

  const updateBaseInfo = (info: BaseFormFields) => {
    baseFormValue.current = info;
  };

  const submitProgress = (
    progressConfig: IWorkFlowStepTemplateReqV1[]
  ): Promise<AxiosResponse<ICreateWorkflowTemplateV1Return>> => {
    return workflow.createWorkflowTemplateV1({
      workflow_template_name: baseFormValue.current?.name,
      desc: baseFormValue.current?.desc,
      instance_name_list: baseFormValue.current?.instanceNameList,
      workflow_step_template_list: progressConfig,
      allow_submit_when_less_audit_level:
        baseFormValue.current?.allowSubmitWhenLessAuditLevel,
    });
  };

  const resetAllForm = () => {
    EventEmitter.emit(EmitterKey.Reset_Workflow_Template_Form);
  };

  return (
    <Card
      title={t('workflowTemplate.create.title.wrapper')}
      extra={[
        <Link to="/progress" key="go-back">
          <Button type="primary">{t('common.back')}</Button>
        </Link>,
      ]}
    >
      <WorkflowTemplateForm
        updateBaseInfo={updateBaseInfo}
        submitProgress={submitProgress}
      >
        <Result
          status="success"
          title={t('workflowTemplate.create.result.title')}
          subTitle={
            <Typography.Link
              type="secondary"
              className="pointer"
              onClick={resetAllForm}
            >
              {t('workflowTemplate.create.result.createNew')}
            </Typography.Link>
          }
        />
        <Row justify="center">
          <Link to="/progress">
            <Button type="primary">
              {t('workflowTemplate.create.result.backToList')}
            </Button>
          </Link>
        </Row>
      </WorkflowTemplateForm>
    </Card>
  );
};

export default CreateWorkflowTemplate;
