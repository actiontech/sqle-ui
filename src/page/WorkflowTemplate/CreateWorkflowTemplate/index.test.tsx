import { shallow } from 'enzyme';
import CreateWorkflowTemplate from '.';
import { WorkFlowStepTemplateReqV1TypeEnum } from '../../../api/common.enum';
import workflow from '../../../api/workflow';
import EmitterKey from '../../../data/EmitterKey';
import { renderWithThemeAndRouter } from '../../../testUtils/customRender';
import {
  mockUseInstance,
  mockUseUsername,
  resolveThreeSecond,
} from '../../../testUtils/mockRequest';
import EventEmitter from '../../../utils/EventEmitter';

describe('CreateWorkflowTemplate', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    mockUseUsername();
    mockUseInstance();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllTimers();
  });

  const mockCreateWorkflowTemplate = () => {
    const spy = jest.spyOn(workflow, 'createWorkflowTemplateV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  test('should match snapshot', async () => {
    const { container } = renderWithThemeAndRouter(<CreateWorkflowTemplate />);
    expect(container).toMatchSnapshot();
  });

  test('should send create workflow template when user submit all form', async () => {
    const createSpy = mockCreateWorkflowTemplate();
    const shallowWrapper = shallow(<CreateWorkflowTemplate />);
    const formWrapper = shallowWrapper.find('WorkflowTemplateForm');
    formWrapper.prop<Function>('updateBaseInfo')({
      name: 'name1',
      desc: 'desc1',
      instanceNameList: ['instanceList1'],
    });
    formWrapper.prop<Function>('submitProgress')([
      {
        assignee_user_name_list: ['name1'],
        desc: 'desc1',
        type: WorkFlowStepTemplateReqV1TypeEnum.sql_review,
      },
      {
        assignee_user_name_list: ['name2'],
        desc: 'desc2',
        type: WorkFlowStepTemplateReqV1TypeEnum.sql_review,
      },
      {
        assignee_user_name_list: ['name2'],
        desc: 'desc2',
        type: WorkFlowStepTemplateReqV1TypeEnum.sql_execute,
      },
    ]);
    expect(createSpy).toBeCalledWith({
      workflow_template_name: 'name1',
      desc: 'desc1',
      instance_name_list: ['instanceList1'],
      workflow_step_template_list: [
        {
          assignee_user_name_list: ['name1'],
          desc: 'desc1',
          type: WorkFlowStepTemplateReqV1TypeEnum.sql_review,
        },
        {
          assignee_user_name_list: ['name2'],
          desc: 'desc2',
          type: WorkFlowStepTemplateReqV1TypeEnum.sql_review,
        },
        {
          assignee_user_name_list: ['name2'],
          desc: 'desc2',
          type: WorkFlowStepTemplateReqV1TypeEnum.sql_execute,
        },
      ],
    });
  });

  test('should dispatch reset form event when user click reset all from button', async () => {
    const emitSpy = jest.spyOn(EventEmitter, 'emit');
    const shallowWrapper = shallow(<CreateWorkflowTemplate />);

    const LinkWrapper = shallow(
      shallowWrapper.find('Result').prop<JSX.Element>('subTitle')
    );
    LinkWrapper.prop<Function>('onClick')();
    expect(emitSpy).toBeCalledWith(EmitterKey.Reset_Workflow_Template_Form);
  });
});
