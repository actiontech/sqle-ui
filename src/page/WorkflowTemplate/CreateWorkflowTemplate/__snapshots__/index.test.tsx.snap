// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWorkflowTemplate should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          workflowTemplate.create.title.wrapper
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/progress"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                common.back
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-space ant-space-vertical full-width-element"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 24px;"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-12 ant-col-offset-6"
            >
              <div
                class="ant-steps ant-steps-horizontal ant-steps-label-horizontal"
              >
                <div
                  class="ant-steps-item ant-steps-item-process ant-steps-item-active"
                >
                  <div
                    class="ant-steps-item-container"
                  >
                    <div
                      class="ant-steps-item-tail"
                    />
                    <div
                      class="ant-steps-item-icon"
                    >
                      <span
                        class="ant-steps-icon"
                      >
                        1
                      </span>
                    </div>
                    <div
                      class="ant-steps-item-content"
                    >
                      <div
                        class="ant-steps-item-title"
                      >
                        workflowTemplate.step.baseFormTitle
                      </div>
                      <div
                        class="ant-steps-item-description"
                      >
                        workflowTemplate.step.baseFormDesc
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-steps-item ant-steps-item-wait"
                >
                  <div
                    class="ant-steps-item-container"
                  >
                    <div
                      class="ant-steps-item-tail"
                    />
                    <div
                      class="ant-steps-item-icon"
                    >
                      <span
                        class="ant-steps-icon"
                      >
                        2
                      </span>
                    </div>
                    <div
                      class="ant-steps-item-content"
                    >
                      <div
                        class="ant-steps-item-title"
                      >
                        workflowTemplate.step.progressTitle
                      </div>
                      <div
                        class="ant-steps-item-description"
                      >
                        workflowTemplate.step.progressDesc
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-steps-item ant-steps-item-wait"
                >
                  <div
                    class="ant-steps-item-container"
                  >
                    <div
                      class="ant-steps-item-tail"
                    />
                    <div
                      class="ant-steps-item-icon"
                    >
                      <span
                        class="ant-steps-icon"
                      >
                        3
                      </span>
                    </div>
                    <div
                      class="ant-steps-item-content"
                    >
                      <div
                        class="ant-steps-item-title"
                      >
                        workflowTemplate.step.resultTitle
                      </div>
                      <div
                        class="ant-steps-item-description"
                      >
                        workflowTemplate.step.resultDesc
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
          style="margin-bottom: 24px;"
        >
          <div
            data-testid="base-form"
          >
            <form
              class="ant-form ant-form-horizontal"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
                >
                  <label
                    class="ant-form-item-required"
                    for="name"
                    title="workflowTemplate.form.label.name"
                  >
                    workflowTemplate.form.label.name
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        id="name"
                        placeholder="common.form.placeholder.input"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
                >
                  <label
                    class=""
                    for="desc"
                    title="workflowTemplate.form.label.desc"
                  >
                    workflowTemplate.form.label.desc
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <textarea
                        class="ant-input textarea-no-resize"
                        id="desc"
                        placeholder="common.form.placeholder.input"
                        rows="3"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item ant-form-item-has-success"
              >
                <div
                  class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
                >
                  <label
                    class=""
                    for="allowSubmitWhenLessAuditLevel"
                    title="workflowTemplate.form.label.allowSubmitWhenLessAuditLevel"
                  >
                    workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-single ant-select-show-arrow"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-activedescendant="allowSubmitWhenLessAuditLevel_list_0"
                              aria-autocomplete="list"
                              aria-controls="allowSubmitWhenLessAuditLevel_list"
                              aria-haspopup="listbox"
                              aria-owns="allowSubmitWhenLessAuditLevel_list"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="allowSubmitWhenLessAuditLevel"
                              readonly=""
                              role="combobox"
                              style="opacity: 0;"
                              type="search"
                              unselectable="on"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-item"
                            title="workflowTemplate.auditLevel.warn"
                          >
                            workflowTemplate.auditLevel.warn
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
                >
                  <label
                    class=""
                    for="instanceNameList"
                    title="workflowTemplate.form.label.instanceNameList"
                  >
                    workflowTemplate.form.label.instanceNameList
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-multiple ant-select-show-search"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <div
                            class="ant-select-selection-overflow"
                          >
                            <div
                              class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                              style="opacity: 1;"
                            >
                              <div
                                class="ant-select-selection-search"
                                style="width: 0px;"
                              >
                                <input
                                  aria-activedescendant="instanceNameList_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="instanceNameList_list"
                                  aria-haspopup="listbox"
                                  aria-owns="instanceNameList_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="instanceNameList"
                                  readonly=""
                                  role="combobox"
                                  style="opacity: 0;"
                                  type="search"
                                  unselectable="on"
                                  value=""
                                />
                                <span
                                  aria-hidden="true"
                                  class="ant-select-selection-search-mirror"
                                >
                                  
                                   
                                </span>
                              </div>
                            </div>
                          </div>
                          <span
                            class="ant-select-selection-placeholder"
                          >
                            common.form.placeholder.select
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
                >
                  <label
                    class="ant-form-item-no-colon"
                    title=" "
                  >
                     
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                          style="margin-right: 8px;"
                        >
                          <button
                            class="ant-btn"
                            type="button"
                          >
                            <span>
                              common.reset
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-space-item"
                        >
                          <button
                            class="ant-btn ant-btn-primary"
                            type="button"
                          >
                            <span>
                              common.nextStep
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="ant-space-item"
          style="margin-bottom: 24px;"
        >
          <div
            data-testid="progress-config"
            hidden=""
          >
            <div
              class="ant-space ant-space-vertical full-width-element"
            >
              <div
                class="ant-space-item"
                style="margin-bottom: 24px;"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-10 ant-col-offset-3"
                  >
                    <div
                      class="ant-steps ant-steps-vertical"
                    >
                      <div
                        class="ant-steps-item ant-steps-item-process ant-steps-item-active"
                      >
                        <div
                          class="ant-steps-item-container"
                        >
                          <div
                            class="ant-steps-item-tail"
                          />
                          <div
                            class="ant-steps-item-icon"
                          >
                            <span
                              class="ant-steps-icon"
                            >
                              1
                            </span>
                          </div>
                          <div
                            class="ant-steps-item-content"
                          >
                            <div
                              class="ant-steps-item-title"
                            >
                              workflowTemplate.progressConfig.createStep.title
                            </div>
                            <div
                              class="ant-steps-item-description"
                            >
                              workflowTemplate.progressConfig.createStep.desc
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ant-steps-item ant-steps-item-process"
                      >
                        <div
                          class="ant-steps-item-container"
                        >
                          <div
                            class="ant-steps-item-tail"
                          />
                          <div
                            class="ant-steps-item-icon"
                          >
                            <span
                              class="ant-steps-icon"
                            >
                              2
                            </span>
                          </div>
                          <div
                            class="ant-steps-item-content"
                          >
                            <div
                              class="ant-steps-item-title"
                            >
                              workflowTemplate.progressConfig.exec.title
                              <div
                                class="ant-steps-item-subtitle"
                                title="workflowTemplate.progressConfig.exec.subTitle"
                              >
                                workflowTemplate.progressConfig.exec.subTitle
                              </div>
                            </div>
                            <div
                              class="ant-steps-item-description"
                            >
                              <div
                                class="ant-row"
                              >
                                <div
                                  class="ant-col ant-col-18"
                                >
                                  <div
                                    class="ant-space ant-space-vertical full-width-element"
                                  >
                                    <div
                                      class="ant-space-item"
                                      style="margin-bottom: 24px;"
                                    >
                                      <div
                                        class="ant-row"
                                      >
                                        <div
                                          class="ant-col ant-col-5"
                                        >
                                          workflowTemplate.form.label.execUser
                                        </div>
                                        <div
                                          class="ant-col ant-col-18"
                                        >
                                          <div
                                            class="ant-select full-width-element ant-select-multiple ant-select-show-search"
                                            data-testid="exec-user-select"
                                          >
                                            <div
                                              class="ant-select-selector"
                                            >
                                              <div
                                                class="ant-select-selection-overflow"
                                              >
                                                <div
                                                  class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                                                  style="opacity: 1;"
                                                >
                                                  <div
                                                    class="ant-select-selection-search"
                                                    style="width: 0px;"
                                                  >
                                                    <input
                                                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                                                      aria-autocomplete="list"
                                                      aria-controls="rc_select_TEST_OR_SSR_list"
                                                      aria-haspopup="listbox"
                                                      aria-owns="rc_select_TEST_OR_SSR_list"
                                                      autocomplete="off"
                                                      class="ant-select-selection-search-input"
                                                      id="rc_select_TEST_OR_SSR"
                                                      readonly=""
                                                      role="combobox"
                                                      style="opacity: 0;"
                                                      type="search"
                                                      unselectable="on"
                                                      value=""
                                                    />
                                                    <span
                                                      aria-hidden="true"
                                                      class="ant-select-selection-search-mirror"
                                                    >
                                                      
                                                       
                                                    </span>
                                                  </div>
                                                </div>
                                              </div>
                                              <span
                                                class="ant-select-selection-placeholder"
                                              >
                                                common.form.placeholder.select
                                              </span>
                                            </div>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                    <div
                                      class="ant-space-item"
                                      style="margin-bottom: 24px;"
                                    >
                                      <div
                                        class="ant-row"
                                      >
                                        <div
                                          class="ant-col ant-col-5 text-black"
                                        >
                                          workflowTemplate.form.label.reviewDesc
                                        </div>
                                        <div
                                          class="ant-col ant-col-18"
                                        >
                                          <textarea
                                            class="ant-input textarea-no-resize"
                                            data-testid="exec-user-desc"
                                            placeholder="common.form.placeholder.input"
                                          />
                                        </div>
                                      </div>
                                    </div>
                                    <div
                                      class="ant-space-item"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col ant-col-7"
                  >
                    <div
                      class="ant-space ant-space-vertical"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 24px;"
                      >
                        <div
                          class="ant-alert ant-alert-warning ant-alert-no-icon"
                          data-show="true"
                          role="alert"
                        >
                          <div
                            class="ant-alert-content"
                          >
                            <div
                              class="ant-alert-message"
                            >
                              <article
                                class="ant-typography clear-margin-bottom"
                              >
                                <h5
                                  class="ant-typography"
                                >
                                  workflowTemplate.progressConfig.ruler.title
                                </h5>
                                <ul
                                  class="clear-margin-bottom"
                                >
                                  <li>
                                    workflowTemplate.progressConfig.ruler.rule1
                                  </li>
                                  <li>
                                    workflowTemplate.progressConfig.ruler.rule2
                                  </li>
                                  <li>
                                    workflowTemplate.progressConfig.ruler.rule3
                                  </li>
                                  <li>
                                    workflowTemplate.progressConfig.ruler.rule4
                                  </li>
                                  <li>
                                    workflowTemplate.progressConfig.ruler.rule5
                                  </li>
                                </ul>
                              </article>
                            </div>
                            <div
                              class="ant-alert-description"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            workflowTemplate.progressConfig.operator.addReview
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-space-item"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-offset-13"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-right: 8px;"
                      >
                        <button
                          class="ant-btn"
                          type="button"
                        >
                          <span>
                            common.prevStep
                          </span>
                        </button>
                      </div>
                      <div
                        class="ant-space-item"
                        style="margin-right: 8px;"
                      >
                        <button
                          class="ant-btn"
                          type="button"
                        >
                          <span>
                            common.reset
                          </span>
                        </button>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.submit
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            data-testid="submit-result"
            hidden=""
          >
            <div
              class="ant-result ant-result-success"
            >
              <div
                class="ant-result-icon"
              >
                <span
                  aria-label="check-circle"
                  class="anticon anticon-check-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="check-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="ant-result-title"
              >
                workflowTemplate.create.result.title
              </div>
              <div
                class="ant-result-subtitle"
              >
                <a
                  class="ant-typography ant-typography-secondary pointer"
                >
                  workflowTemplate.create.result.createNew
                </a>
              </div>
            </div>
            <div
              class="ant-row ant-row-center"
            >
              <a
                href="/progress"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                    workflowTemplate.create.result.backToList
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
